packages/twenty-front/src/modules/favorites/hooks/useCreateFavoriteFolder.ts

```diff
@@ -0,0 +1,32 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useCreateOneRecord } from '@/object-record/hooks/useCreateOneRecord';
+import { usePrefetchedFavoritesData } from './usePrefetchedFavoritesData';
+import { usePrefetchedFavoritesFoldersData } from './usePrefetchedFavoritesFoldersData';
+
+export const useCreateFavoriteFolder = () => {
+  const { createOneRecord: createFavoriteFolder } = useCreateOneRecord({
+    objectNameSingular: CoreObjectNameSingular.FavoriteFolder,
+  });
+
+  const { currentWorkspaceMemberId } = usePrefetchedFavoritesData();
+  const { favoriteFolders } = usePrefetchedFavoritesFoldersData();
+
+  const createNewFavoriteFolder = async (name: string): Promise<void> => {
+    if (!name || !currentWorkspaceMemberId) {
+      return;
+    }
+
+    const maxPosition = Math.max(
+      ...favoriteFolders.map((folder) => folder.position),
+      0,
+    );
+
+    await createFavoriteFolder({
+      workspaceMemberId: currentWorkspaceMemberId,
+      name,
+      position: maxPosition + 1,
+    });
+  };
+
+  return createNewFavoriteFolder;
```

Comment:
even when we only return one function / object we do return { createNewFavoriteFolder }, this enforce consistency accross the codebase and avoid useless refactoring in case we start returning multiple objects